### PR TITLE
Time management

### PR DESCRIPTION
**Pull request by kryczkal**
The main algorithm is laid out - it is currently working, suboptimally, but nevertheless

Except for the time allocation function, everything else works:
- Time manager thread
- Custom engine command for timePerMove calculation given the current state of the board
- Debug tools for the time algorithm:
    a) tooling to plot the time managment function across a game.

From this point onward I will come up with, create and test out different time management functions.

### CODE REVIEW

**Comment by kryczkal** [comment]
I forgot to clang-format - will happen in a sec

File: Tests/GameTimeManagerTimePlot/GameTimeManagerTimePlot.py
Comment (Jlisowskyy) [inline Tests/GameTimeManagerTimePlot/GameTimeManagerTimePlot.py:1]: In future try to make test scripts and functions as modular as possible, especially when there is no big cost associated with it. In this particular example you could read moves from the comand line arguments or something. I think its beneficial in longer run

File: include/ThreadManagement/GameTimeManagerUtils.h
Comment (Jlisowskyy) [inline include/ThreadManagement/GameTimeManagerUtils.h:1]: Its totally no necessary to implement everything as constexpr, but I appreciate the effort spend on it.

**Review by Jlisowskyy — APPROVED** [review]
Good job

**Pull request merged**